src/Entity/User.php

```diff
@@ -314,4 +314,9 @@ public function toArray(): array
     {
         return get_object_vars($this);
     }
+
+    public function isNewUser(): bool
+    {
+        return $this->id === null;
```

Comment:
In the Field, we check if a field is new like so: https://github.com/bolt/core/blob/577f5b9fff286c7a6ffd81553bfb903e8d911e58/src/Entity/Field.php#L206

Do we have to update that here, or there? Given the strict comparison operator (`===`), I wonder if both cases work?

Reply:
well in my case I can say, that creating a new user produces a $user->id of null, not 0
Thats why I used that here.
Could be that the Field Entity is handeld differently so a new field has an ID of 0 before it is persisted

Reply:
Ah, yes you are correct.

https://github.com/bolt/core/blob/2d0a6017ad385c4842d0431b235a78ca36933c23/src/Entity/Field.php#L51

By the looks of it, we explicitly set it to 0 for the field 🤷‍♂️

All is good then. 👍